### ISSUE TITLE
Error: [Model] error loading weights.

### ISSUE THREAD

**Issue by calclavia** [transcranial/keras-js#46]
I'm running the latest Keras JS on a trained Keras model. Following the instructions given on the README, I tested loading my model before training (random initialized weights) and it works.

However, after I perform training and use the exact same way to called `model.save_weights`, Keras JS is unable to read the model file. When trying to parse the metadata, it gives `[Model] error loading weights.`

Any insight into why this would happen?

**Comment by stekaiser** [comment]
Which Keras version did you use for training? 

Could you provide a minimal working example which reproduces your issue (via jsfiddle or as GitHub gist)?

**Comment by calclavia** [comment]
I'm using Keras 1.2.1.

Here are my model JSON files.
https://gist.github.com/calclavia/0939ea8e8f2a5b7374927415ee786f64

Here's the protobuf weights file, as generated from the `encoder.py`.
https://www.dropbox.com/s/82yydvye79wmm4g/model_weights.buf?dl=0

**Comment by stekaiser** [comment]
That looks good. I think I cannot help you on this issue.

Have you tried an older Keras version? Some time ago a change in Keras broke Keras.js.

**Comment by transcranial** [comment]
@calclavia can you also post the weights metadata file (`ai_path + '_metadata.json'`)? That error is due to not being able to find the corresponding weights metadata parameters. The `weights.buf` file is just all the raw weights compacted together, and the `metadata.json` file tells the model how to split the arraybuffer across all the layers during initialization.

**Comment by calclavia** [comment]
@transcranial I've already posted it in the gist. It's at the bottom:
https://gist.github.com/calclavia/0939ea8e8f2a5b7374927415ee786f64#file-model_metadata-json

**Comment by transcranial** [comment]
@calclavia didn't see that, thanks. So the issue is that the batchnorm layers didn't make it into your `model_metadata.json` on running `encoder.py`, which is pretty strange.

I can't reproduce when I take your `model.json` config file and do the following (keras 1.2.2, tensorflow 1.0):

```py
from keras.models import model_from_json
with open('model.json', 'r') as f:
    config = f.read()
model = model_from_json(config)
model.save_weights('test.hdf5')
```

```sh
$ python encoder.py test.hdf5
```

The resulting `test_metadata.json` has the batchnorm layers info, which loads and works fine. What does your hdf5 weights file look like?

**Comment by calclavia** [comment]
@transcranial  My `model.h5` file is here: https://www.dropbox.com/s/4qcsx3hw109ifij/model.h5?dl=0

**Comment by transcranial** [comment]
Yeah, your hdf5 file doesn't have weights from any of your BatchNormalization layers. You should look into that.

**Comment by calclavia** [comment]
@transcranial Oh, I think I found the problem... Removing the batch norm seems to allowed to work. I think I accidentally modified my model architecture before outputting the `model.json` file. Thanks for the help! My bad. :)

**Comment by herosunly** [comment]
import h5py
hdf5_file = h5py.File('my_model_weights', mode='r')
print(list(hdf5_file))
My result is [u'activation_1', u'dense_1', u'dropout_1', u'embedding_1', u'lstm_1'], what is wrong with my weights file?
